lib/_http_outgoing.js

```diff
@@ -487,7 +492,7 @@ OutgoingMessage.prototype.end = function(data, encoding, callback) {
     encoding = null;
   }
 
-  if (data && typeof data !== 'string' && !(data instanceof Buffer)) {
+  if (data && !(typeof data === 'string' || data instanceof Buffer)) {
```

Comment:
This was less of a style change and more of an optimization, leaving the `data instanceof Buffer` to not be called if it was a string. Would you still like them removed?